### ISSUE TITLE
Auto generate nav based on pages Headers 2

### ISSUE THREAD

**Issue by ia3andy** [quarkiverse/quarkiverse-docs#102]
Maybe there is an Asciidoc macro or something that will generate the TOC based on the file. I know there is this: https://docs.asciidoctor.org/asciidoc/latest/toc/ but it doesn't seem to work for XREF links

_Originally posted by @gastaldi in https://github.com/quarkiverse/quarkus-renarde/pull/148#discussion_r1266801249_

**Comment by holly-cummins** [comment]
Isn't this done already? 

See, for example, https://docs.quarkiverse.io/quarkus-pact/dev/consumer-index.html. I did a quick code search, and all the content in the left is only based on headers, not on manual config.

<img width="1502" alt="image" src="https://github.com/quarkiverse/quarkiverse-docs/assets/11509290/f12eb2d7-b86c-4ad9-a695-4e48c403abd0">

**Comment by holly-cummins** [comment]
Or is the idea to also have that content on the right appear on the left? I think that's just a layout thing, not a generation thing, but I'd need to experiment. 

Maybe @ia3andy you could provide a screenshot of the intended effect? I looked at https://github.com/quarkiverse/quarkus-renarde/pull/148#discussion_r1266801249 but it wasn't clear enough to me where the nav was, and what was different from the current state.

**Comment by ia3andy** [comment]
@holly-cummins so as a User point of view, it seems to me that:
- Left nav is top level main menu/nav (currently manual)
- Right nav is page deep summary/nav (automatic)

In the left Nav, I would automatically link to all pages and reference h2 (main titles)
In the right Nav, keep it as it is.

**Comment by holly-cummins** [comment]
Do you think there's value in having stuff over in the right that duplicates stuff on the left? It feels like ... a lot of duplication. 

I know it's how Antora pages are usually structured, but that doesn't make it good. :) I wonder if just having everything on the left would be more natural to people who weren't antora-users.

On the left nav, automatically discovering the pages may need some thought, to make sure we 
- don't break existing navs
- do the right thing

We can apply a convention that `index.adoc` is the top level, but if there's multiple other adocs, we may need to do something clever to order them. I guess requiring the adoc pages to be numbered would sort it out. 

I also don't know if antora lets us drop the nav element in antora.yml, but I can do some experiments. 

I agree that when viewing an extension, the left bar is pretty sparse, and I definitely agree that filling it in manually down to H2 is (a) boring and (b) a recipe for errors.

**Comment by ia3andy** [comment]
I think it looks nice: https://docs.quarkiverse.io/quarkus-renarde/dev/

**Comment by holly-cummins** [comment]
I agree that the richer navigation on the left is *much* better than the default. I'm not sure what value the navigation on the right is bringing. It means that to navigate, a user has to flip their attention back and forth between the left and right bars, and a lot of the screen real estate is taken up with repeated content. 

I'd suggest, rather than this:

![image](https://github.com/quarkiverse/quarkiverse-docs/assets/11509290/d0b8769a-da86-46a2-b6c9-ede2f5483891)

... something like this:

<img width="1628" alt="image" src="https://github.com/quarkiverse/quarkiverse-docs/assets/11509290/ca780d57-1b57-4a9d-b4f1-d5e37e862e23">

We could add some vertical bar or other divider, and perhaps even some sort of toggle or slider for how many levels to show.

**Comment by holly-cummins** [comment]
I've just done a quick unscientific survey to see how common the right-bar pattern is. It's the Antora default, which does make it reasonably common. 

For example, AWS have it, but they've relabelled the header from 'Content' (which is a totally information-free header) to 'On this page' (which at least gives some indication why there are two nav bars)

<img width="1624" alt="image" src="https://github.com/quarkiverse/quarkiverse-docs/assets/11509290/5df08424-2245-459a-82cb-b5ed722dc44a">

Apple do not have two navbars in their developer documentation :) 

<img width="1624" alt="image" src="https://github.com/quarkiverse/quarkiverse-docs/assets/11509290/b624364b-2730-4221-83d4-9c6ac0b2f428">

Microsoft have one navbar and use the right-hand space for extra resources, so that they stay pinned to the top of the page: 

<img width="1624" alt="image" src="https://github.com/quarkiverse/quarkiverse-docs/assets/11509290/1ccab1b6-ce00-4caa-be2b-60dd58272ee2">

Our normal guides have nav on the right, but they *do not* have nav on the left. That's another option we could do:

<img width="1624" alt="image" src="https://github.com/quarkiverse/quarkiverse-docs/assets/11509290/49724a24-8b65-4670-9f4f-a02e3f51377a">

**Comment by ia3andy** [comment]
@gastaldi @unfromage

The left doc is folded and is acting as shortcut to the main section of the doc, it also help split the content rather than having one huge monolyth.

The right doc is a summary of the content and follow the cursor while scrolling to help navigate it.

**Comment by FroMage** [comment]
> I agree that the richer navigation on the left is much better than the default. I'm not sure what value the navigation on the right is bringing. It means that to navigate, a user has to flip their attention back and forth between the left and right bars, and a lot of the screen real estate is taken up with repeated content.

Ah, totally agree :)

> The left nav is folded and is acting as shortcut to the main sections of the doc, it also helps split the content rather than having one huge monolith.

> The right nav is a summary of the content and follow the cursor while scrolling to help navigate it.

Spliting is fine. But in my experience the left is not folded, and both follow you. I don't find it helpful to split navigation into two parts arbitrarily.
